apiserver/params/storage.go

```diff
@@ -544,21 +554,42 @@ type VolumeDetails struct {
 	// VolumeTag is the tag for the volume.
 	VolumeTag string `json:"volume-tag"`
 
+	// BindingTag is the tag of an entity to which the
+	// volume's lifetime is bound.
+	BindingTag string `json:"binding-tag,omitempty"`
+
 	// Info contains information about the volume.
 	Info VolumeInfo `json:"info"`
 
+	// Life contains the lifecycle state of the volume.
+	// Old versions of the API do not populate this field,
+	// so it may be omitted.
+	Life Life `json:"life,omitempty"`
+
 	// Status contains the status of the volume.
 	Status EntityStatus `json:"status"`
 
 	// MachineAttachments contains a mapping from
 	// machine tag to volume attachment information.
-	MachineAttachments map[string]VolumeAttachmentInfo `json:"machine-attachments,omitempty"`
+	MachineAttachments map[string]VolumeAttachmentDetails `json:"machine-attachments,omitempty"`
 
 	// Storage contains details about the storage instance
 	// that the volume is assigned to, if any.
 	Storage *StorageDetails `json:"storage,omitempty"`
 }
 
+// VolumeAttachmentDetails describes a volume attachment.
+type VolumeAttachmentDetails struct {
+	// NOTE(axw) for backwards-compatibility, this must not be given a
+	// json tag. This ensures that we collapse VolumeAttachmentInfo.
+	VolumeAttachmentInfo
```

Comment:
can we add a todo here to remove this compatibility struct when we get a chance to rev up the api
here and below